onnxruntime/core/providers/xnnpack/xnnpack_execution_provider.cc

```diff
@@ -162,7 +162,16 @@ XnnpackExecutionProvider::XnnpackExecutionProvider(const XnnpackExecutionProvide
                                  (info.session_options &&
                                   info.session_options->config_options.GetConfigOrDefault(
                                       kOrtSessionOptionsConfigAllowIntraOpSpinning, "1") == "1");
-  if (xnn_thread_pool_size > 1 && allow_intra_op_spinning && ort_thread_pool_size > 1) {
+  if (xnn_thread_pool_size == 0) {
```

Comment:
pthread_pool will reset the thread_nums by default if xnn_thread_pool_size == 0
https://github.com/Maratyszcza/pthreadpool/blob/560c60d342a76076f0557a3946924c6478470044/src/pthreads.c#L230

Reply:
If xnnpack is doing a good job of sizing the threadpool itself (logic seems similar to what ORT is doing and using cpuinfo if possible) do we need to make this change?